### PR TITLE
Refactor `use_tic()`

### PR DESCRIPTION
**Pull request by pat-s**
# Changes

- Add new templates: Asks the user what to build (single/matrix, Linux/macOS), which CI provider to use at all and if multiple R versions are desired

- Update templates 
  - install pkgs in parallel
  - use ccache for faster re-installs of compiled code

- move some functions to their own .R file to clean up a bit

- update vignette

- use the _cli_ package (dev version) for more sophisticated cli output

- removes the `after_install` stage as this stage is deprecated by Travis

- adds some info to the tic.R template

- update badges
  - add codecov badge
  - use shield.io badges

- export `use_*_yml()` functions so that users can install yml templates without `use_tic()`

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -11,7 +11,6 @@ before_install:
   - R -q -e 'remotes::install_local(getwd(), force = TRUE); print(tic::dsl_load()); tic::prepare_all_stages()'
   - R -q -e 'tic::before_install()'
 install: R -q -e 'tic::install()'
-after_install: R -q -e 'tic::after_install()'
```
Comment (krlmlr) [inline .travis.yml]: Is this intended?

Comment (pat-s) [reply .travis.yml]: This stage is not supported (anymore) by Travis. Discovered this by running the official Travis lintr.

Comment (pat-s) [reply .travis.yml]: > Maintaining multiple .yml templates for each CI platform seems like a nightmare in the long run. Is there any chance to generalize this?

We could trim it down to one file for "build", "deploy" and "meta" (matrix, libs) and cat them together when building the template. This way, we would only have to maintain three files at max per CI provider.

Comment (krlmlr) [reply .travis.yml]: Sounds good.

**Comment by krlmlr** [comment]
Thanks!

**Pull request merged**